tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -194,6 +196,104 @@ def test_non_org_rename_403(self):
         self.get_error_response(other_org.slug, "old_slug", status_code=403)
 
 
+class ProjectUpdateTestTokenAuthenticated(APITestCase):
+    endpoint = "sentry-api-0-project-details"
+    method = "put"
+
+    def setUp(self):
+        super().setUp()
+        self.org_slug = self.project.organization.slug
+        self.proj_slug = self.project.slug
+
+    def test_member_updates_denied_with_token(self):
+        project = self.create_project(platform="javascript")
+        user = self.create_user("bar@example.com")
+        self.create_member(
+            user=user,
+            organization=project.organization,
+            teams=[project.teams.first()],
+            role="member",
+        )
+
+        token = ApiToken.objects.create(user=user, scope_list=["project:write"])
+        authorization = f"Bearer {token.token}"
+
+        data = {"platform": "rust"}
+
+        url = reverse(
+            "sentry-api-0-project-details",
+            kwargs={"organization_slug": project.organization.slug, "project_slug": project.slug},
+        )
+        response = self.client.put(url, format="json", HTTP_AUTHORIZATION=authorization, data=data)
```

Comment:
You can use `self.get_error_response(status_code=status.HTTP_403_FORBIDDEN, format=.....)`

Reply:
Oh actually, @RyanSkonnord helped me figure this one out. :) To use the `get_error_response` we have to call `self.login_as(user)` which uses session authentication, but we explicitly need to use token-based auth for these tests.